run-queue

```diff
@@ -191,11 +196,22 @@ def main():
                     return 1
 
             if isinstance(cmd, list):
-                cmd_str = ' '.join(cmd)
+                cmd_str = shlex.join(cmd)
             else:
                 cmd_str = cmd
+                cmd = ['sh', '-ec', cmd_str]
+            env = os.environ.copy()
             logging.info("Consuming message with command: %s", cmd_str)
-            p = subprocess.Popen(cmd, shell=isinstance(cmd, str))
+
+            if opts.wrapper:
+                cmd.insert(0, opts.wrapper)
+                env['TASK_QUEUE'] = queue_name
+                try:
+                    env['TASK_CONTAINER'] = body['container']
```

Comment:
`body['container']` may well be `None`, which is not a valid value for an environment variable.

How about more like:

```python
if container := body.get('container'):
    env['TASK_CONTAINER'] = container
```

or more defensively:

```python
container = body.get('container')
if isinstance(container, str):
    env['TASK_CONTAINER'] = container
```

Reply:
I take it you mean it may be `null`?

Reply:
Done.